pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,718 @@
+
+    @b_vector_name.setter
+    def b_vector_name(self, val):
+        if val in self.b_vector_choices:
+            self.__oarray.SetPropValue("B Vector", val)
+        else:
+            self.logger.warning("B vector name not available")
+
+    @property
+    def a_size(self):
+        """Number of cells in the vector A direction."""
+        return int(self.__app.get_oo_property_value(self.__omodel, self.name, "A Cell Count"))
+
+    @a_size.setter
+    def a_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("A Cell Count", val)
+        self.logger.warning("A size cannot be modified.")
+
+    @property
+    def b_size(self):
+        """Number of cells in the vector B direction."""
+        return int(self.__app.get_oo_property_value(self.__omodel, self.name, "B Cell Count"))
+
+    @b_size.setter
+    def b_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("B Cell Count", val)
+        self.logger.warning("B size cannot be modified.")
+
+    @property
+    def padding_cells(self):
+        """Number of padding cells."""
+        return int(self.__app.get_oo_property_value(self.__omodel, self.name, "Padding"))
+
+    @padding_cells.setter
+    def padding_cells(self, val):
+        self.__oarray.SetPropValue("Padding", val)
+
+    @property
+    def coordinate_system(self):
+        """Coordinate system name."""
+        cs_dict = self.__map_coordinate_system_to_id()
+        res = "Global"
+        for name, cs_id in cs_dict.items():
+            if cs_id == self.__cs_id:
+                res = name
+        if res == "Global":
+            self.logger.warning("Coordinate system is not loaded. Save the project.")
+        return res
+
+    @coordinate_system.setter
+    def coordinate_system(self, name):
+        cs_dict = self.__map_coordinate_system_to_id()
+        if name not in cs_dict:
+            self.logger.warning("Coordinate system is not loaded. Save the project.")
+        else:
+            self.__cs_id = cs_dict[name]
+            self.edit_array()
+
+    @pyaedt_function_handler()
+    def update_properties(self):
+        """Update component array properties.
+
+        Returns
+        -------
+        dict
+           Ordered dictionary of the properties of the component array.
+        """
+        # From 2024R1, array information can be loaded from a CSV, and this method is not needed.
+        if self.__app.settings.aedt_version > "2023.2":  # pragma: no cover
+            self.__array_info_path = self.export_array_info(array_path=None)
+        else:
+            self.__app.save_project()
+        new_properties = self.properties
+        if len(new_properties["component"]) != len(self.post_processing_cells):
+            self.__post_processing_cells = {}
+            new_properties = self.properties
```

Comment:
Not sure about theses locs. Is it to ensure that we are resilient with the fact that a user has deleted its csv file ?

Reply:
It is for the case that the user remove one 3DComponent that belongs to the array, It is tricky because post processing property can not retrieved from the AEDT file or the CSV, so we need to reset the information, and we need this update_properties method. I have submitted a bug internally to get all this information.

Reply:
Ok. In that case, I would recommend leaving a TODO commentary to remember to update this locs 👍